src/openvic-simulation/economy/FactoryProducer.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <cstdint>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+	class FactoryProducer final {
+	private:
+		static constexpr uint8_t DAYS_OF_HISTORY = 7;
+		uint8_t PROPERTY(profit_history_current);
+		std::array<fixed_point_t, DAYS_OF_HISTORY> PROPERTY(daily_profit_history);
+		ProductionType const& PROPERTY(production_type);
+		fixed_point_t PROPERTY(revenue_yesterday);
+		fixed_point_t PROPERTY(output_quantity_yesterday);
+		fixed_point_t PROPERTY(unsold_quantity_yesterday);
+		fixed_point_t PROPERTY(size_multiplier);
+		ordered_map<Pop*, Pop::pop_size_t> PROPERTY(employees);
+		Good::good_map_t PROPERTY(stockpile);
+		fixed_point_t PROPERTY(budget);
+		fixed_point_t PROPERTY(balance_yesterday);
+		fixed_point_t PROPERTY(received_investments_yesterday);
+		fixed_point_t PROPERTY(market_spendings_yesterday);
+		fixed_point_t PROPERTY(paychecks_yesterday);
+		uint32_t PROPERTY(unprofitable_days);
+		uint32_t PROPERTY(injected_days);
+		uint32_t PROPERTY(days_without_input);
+		uint8_t PROPERTY_RW(hiring_priority);
+
+	public:
+		FactoryProducer(
+			ProductionType const& new_production_type, const fixed_point_t new_size_multiplier,
+			const fixed_point_t new_revenue_yesterday, const fixed_point_t new_output_quantity_yesterday,
+			const fixed_point_t new_unsold_quantity_yesterday, ordered_map<Pop*, Pop::pop_size_t>&& new_employees,
+			Good::good_map_t&& new_stockpile, fixed_point_t new_budget, const fixed_point_t new_balance_yesterday,
+			const fixed_point_t new_received_investments_yesterday, const fixed_point_t new_market_spendings_yesterday,
+			const fixed_point_t new_paychecks_yesterday, const uint32_t new_unprofitable_days, const uint32_t new_injected_days,
+			const uint32_t new_days_without_input, const uint8_t new_hiring_priority, const uint8_t new_profit_history_current,
+			std::array<fixed_point_t, DAYS_OF_HISTORY>&& new_d
```

Comment:
```suggestion
			std::array<fixed_point_t, DAYS_OF_HISTORY>&& new_daily_profit_history
```
Argument name cut short